src/_cffi_src/commoncrypto/secure_transport.py

```diff
@@ -0,0 +1,315 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+INCLUDES = """
+#include <Security/SecureTransport.h>
+"""
+
+TYPES = """
+typedef ... *SSLContextRef;
+typedef const void *SSLConnectionRef;
```

Comment:
Ok.

On Sat, Jul 23, 2016 at 9:19 AM, Cory Benfield <notifications@github.com>
wrote:

> In src/_cffi_src/commoncrypto/secure_transport.py
> <https://github.com/pyca/cryptography/pull/3054#discussion_r71972392>:
>
> > @@ -0,0 +1,315 @@
> > +# This file is dual licensed under the terms of the Apache License, Version
> > +# 2.0, and the BSD License. See the LICENSE file in the root of this repository
> > +# for complete details.
> > +
> > +from __future__ import absolute_import, division, print_function
> > +
> > +INCLUDES = """
> > +#include <Security/SecureTransport.h>
> > +"""
> > +
> > +TYPES = """
> > +typedef ... *SSLContextRef;
> > +typedef const void *SSLConnectionRef;
>
> No.
>
> The purpose of this is to be a plain object that is passed into/out of
> secure transport, so ideally you'd use it with ffi.to_handle and
> ffi.from_handle. If you change this to a ... typedef, CFFI complains:
>
> Traceback (most recent call last):
>   File "/Users/cory/tmp/securetransportscratch/src/securetransport/low_level.py", line 51, in python_write_func
>     context = ffi.from_handle(connection)
> TypeError: expected a 'cdata' object with a 'void *' out of new_handle(), got 'struct $$SSLConnectionRef *'
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pyca/cryptography/pull/3054/files/651d9c9ea0aaeabdeec81d4daaddf800a3203c5f#r71972392>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAADBB0E2f9ZvbQWQiTJe-0Hp1dYc3FSks5qYhTPgaJpZM4JSw_3>
> .
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: D1B3 ADC0 E023 8CA6